sentry-android-replay/src/main/java/io/sentry/android/replay/ReplayIntegration.kt

```diff
@@ -388,10 +415,46 @@ class ReplayIntegration(
             return
         }
 
+        try {
+            context.unregisterComponentCallbacks(this)
+        } catch (ignored: Throwable) {
+        }
         stop()
         replayExecutor.gracefullyShutdown(options)
     }
 
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        if (!isEnabled.get()) {
+            return
+        }
+
+        recorder?.stopRecording()
```

Comment:
I'm wondering if `onConfigurationChanged` could be triggered without the width/height actually changing? Would it make sense to guard this the following way ~ `if (oldConfig == newConfig) { return }`

Reply:
yeah it can be triggered in the following cases 
![image](https://github.com/getsentry/sentry-java/assets/4999776/0686a4ec-2960-44cc-a92f-815cea3d2a50)

we're mostly interested in the first two, but I guess it's fine to stop and restart the recording for the others as well, just to make sure we don't try to encode different configurations into the same video segment, wdyt? Otherwise, we'd probably have to check what actually changed in the configuration, and if it's not window/screen size, then we do nothing.

Reply:
Yeah it makes sense to me to create a new segment, let's keep it simple and check later if too many config changes are causing an issue 👍  (e.g. keyboard show/hide, network provider changing, ...)

Reply:
sounds good! I guess we'd need some kind of internal tracking for that, but that's for later